src/scripting/scriptingService.ts

```diff
@@ -55,11 +55,27 @@ export class ScriptingService {
         return scriptingObject;
     }
 
-    public async scriptSelect(node: TreeNodeInfo, uri: string): Promise<string> {
+    /**
+     * Helper to create scripting params
+     */
+    public createScriptingParams(node: TreeNodeInfo, uri: string, operation: ScriptOperation): ScriptingParams {
         const scriptingObject = this.getObjectFromNode(node);
         let serverInfo = this._connectionManager.getServerInfo(node.connectionCredentials);
+        let scriptCreateDropOption: string;
+        switch (operation) {
+            case (ScriptOperation.Select):
+                scriptCreateDropOption = 'ScriptSelect';
```

Comment:
Bleh - why is this not an enum. Anyways I know this is just a string value on the service side but I would suggest making either a string enum or a string literal type to reduce the chance of errors in the future.